### ISSUE TITLE
Title bar is not blurred in KDE Plasma 5.25 (Beta)

### ISSUE THREAD

**Issue by pnedyalkov91** [Luwx/Lightly#171]
Yesterday I updated my Plasma to 5.25 (Beta) and noticed the title bar is not blurred. I haven't had any issues when I use Plasma 5.24.

**Configuration:**
Operating System: Arch Linux
KDE Plasma Version: 5.24.90
KDE Frameworks Version: 5.94.0
Qt Version: 5.15.4
Kernel Version: 5.17.9-arch1-1 (64-bit)
Graphics Platform: X11
Lightly version: The latest commit from git

Screenshot:
![image](https://user-images.githubusercontent.com/59796383/169452556-208dbef2-08d7-4c34-b342-0c57ac308504.png)

**Comment by plashenkov** [comment]
Officially 5.25 is out now, and yes — titlebars aren't blurred anymore.

**Comment by adam-adrian** [comment]
AFAIK, you must use setBlurRegion API on the code to make blur enabled, like ClassiK: https://github.com/paulmcauley/classik/commit/fe48c052f529e889706214c9af3db200fbfdb2a8

**Comment by adam-adrian** [comment]
But weirdly, in my case, Konsole is still blurred
![Screenshot_20220525_123725](https://user-images.githubusercontent.com/96514212/174202072-ef9255a5-ace5-4fcc-b4a1-2bf443b9afc2.png)

**Comment by plashenkov** [comment]
Yup, I've noteced as well that Konsole is perfectly blurred. It uses older version of Qt widgetset maybe? Just an assumption.

**Comment by adam-adrian** [comment]
> Yup, I've noteced as well that Konsole is perfectly blurred. It uses older version of Qt widgetset maybe? Just an assumption.

Maybe?
We just can waiting for @Luwx to implement it or someone making a pull request for it.

**Comment by plashenkov** [comment]
@adam-adrian Looks very nice. Which color scheme do you use to have the dark theme?

**Comment by adam-adrian** [comment]
Willow dark glass

**Comment by plashenkov** [comment]
@adam-adrian
And I can see a nice lighter bevel around your windows. Looks great, how have you got this?
I use https://github.com/a-parhom/LightlyShaders to workaround corners bug, but it doesn't provide such nice bevel (AFAIK).

**Comment by adam-adrian** [comment]
I don't know, maybe from the lightly itself

**Comment by pnedyalkov91** [comment]
I've noticed a weird situation. Konsole, Dolphin, and System Settings have blurred title bars but when I try to resize windows, the title bar blur has gone...

**Comment by doncsugar** [comment]
I believe the reason that Konsole seems to be blurred is because blur was manually enabled. Without shaders, Konsole has Korners because a blurred rectangle is being drawn behind the whole window. Blur has been broken in Konsole for a while now. I guess it just wasn't noticed because blur was enabled manually. If you disable the option to blur Konsole, you will probably see this:
![image](https://user-images.githubusercontent.com/46570709/174448279-beb9516e-73f9-4260-8d80-59a6c9da20e6.png)



@plashenkov Have you checked the settings for the shaders in System Settings > Workspace Behavior > Desktop Effects? From there you can turn the bevel on.

**Comment by plashenkov** [comment]
@doncsugar Yup, I've already found it. It was hard since there's no info about how to open LightlyShaders config in [Luwx/LightlyShaders](https://github.com/Luwx/LightlyShaders) or [a-parhom/LightlyShaders](https://github.com/a-parhom/LightlyShaders) readme :)

Thanks for your Willow theme indeed! I've found Lightly and LightlyShaders thanks to you.

**Comment by doncsugar** [comment]
@plashenkov You're welcome! Of course, I couldn't have made Willow if Luwx hadn't taken the time to made Lightly!

**Comment by daxid** [comment]
Missing blur in tittle bars is surely killing the vibe :-(

Anybody on this ?

Thanks @adam-adrian  for your input (pretty much the only comment related to the original issue here), but I could not understand what to make with it...

Best regards

**Comment by boehs** [comment]
> Anybody on this ?

https://github.com/Luwx/Lightly/pull/190

Also see #188

**Comment by plashenkov** [comment]
Hey @boehs, that's great, thank you! Your version works with 5.25

**Comment by daxid** [comment]
Absolutely, I just built it and the blur vibe is back :-)

Good enough for now. Thanks a ton @boehs

**Comment by plashenkov** [comment]
@boehs
Sorry, I cannot create issues in your repo.

Please consider adding dark color scheme. This one looks nice:
https://github.com/Luwx/Lightly/blob/73739ab9eb6934ce886d4d237539208556fe8902/Darkly.colors
It can be used as a common color scheme or for some specific applications (Konsole, for example).

**Comment by boehs** [comment]
> Sorry, I cannot create issues in your repo.

Fixed

> Please consider adding dark color scheme. This one looks nice:

Will do today

**Comment by Lumipyry** [comment]
**IF YOU WANT TO HAVE BLURRED TITLEBAR IN PLASMA 5.25**
From now one there could be two versions of Lightly in Plasma 5.25: One with fully opaque titlebar and one with blurred titlebar!

You all may have noticed that michaelwalld has solved the issue of  titlebar not being blurred in Plasma 5.25 and has made a pull request on it: https://github.com/Luwx/Lightly/pull/190

At the moment, if you want to have titlebars blurred in Plasma 5.25 - uninstall Lightly and re-install it from Michael's fork: https://github.com/michaelwalld/Lightly

I'd like to add, that after first shock, **a fully opaque titlebar in Plasma 5.25 is actually quite beautiful**. So I suggest that **from now on there could be two versions of Lightly: One with fully opaque titlebar and one with blurred titlebar**.....

**Comment by boehs** [comment]
@Lumipyry on my fork I've merged several additional patches (https://github.com/boehs/lightly) and will continue to do so until

1. Luwx comes back
2. Someone begins maintaining this repository on their behalf

**Comment by plashenkov** [comment]
@boehs I'm using your version and encounter visual glitches where there are blurred areas. For example, if I open Dolphin, everything is okay, but when I open another app's window, or if I open some menus, glitches appear on the Dolphin's window titlebar. Did you see something like that?

![Screenshot_20221127_052551___2](https://user-images.githubusercontent.com/2331932/204117402-0b24a150-0286-4ad5-a852-bd1790322a53.png)

![Screenshot_20221127_052639___2](https://user-images.githubusercontent.com/2331932/204117407-e45e2e3a-ed87-4be3-8641-bd20fe47a795.png)

**Comment by boehs** [comment]
> Did you see something like that?

No, but I recall seeing some issue opened here similar to that. Let me look around. What distro are you using?

**Comment by plashenkov** [comment]
@boehs I'm using KDE neon 5.26 (based on Ubuntu 20.04) with all latest updates.
